src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -465,7 +465,6 @@ private void CancelRequestAbortedTokenCallback()
                 if (_abortedCts != null && !_preventRequestAbortedCancellation)
                 {
                     localAbortCts = _abortedCts;
-                    _abortedCts = null;
```

Comment:
Note that this change means the CTS isn't cleared on abort. It will be disposed when the HttpProtocol type is reset.

Reply:
If this is a problem, a separate field could be added to save the CT while clearing the CTS.

Reply:
I think this is an improvement rather than a problem. The reason we couldn't dispose it here is because we didn't want to get ODEs when middleware accessed it when it was still running. Reset() is only called between middleware invocations.